src/pocketmine/entity/Living.php

```diff
@@ -786,10 +787,44 @@ public function getTargetBlock(int $maxDistance, array $transparent = []){
 			}
 		}catch(\ArrayOutOfBoundsException $e){
 		}
-
 		return null;
 	}
 
+        /**
+         *
+         * For the exact entity use a precision >= 0.998
+         *
+         * @param AxisAlignedBB|null $distance
+         * @param float              $precision
+         *
+         * @return null|Entity
+         *
+         */
+        public function getEntityLookingAt(AxisAlignedBB $distance = null, float $precision = 0.998){
+                if($distance === null){
+                        $pointA = $this->subtract(100, 100, 100);
+                        $pointB = $this->add(100, 100, 100);
```

Comment:
??????? Why on earth ???????

It should be more than sufficient to give a floating-point linear distance, and then multiply the entity's direction vector by it to obtain points A and B:

```php
$pointA = $this->asVector3();
$pointB = $this->getDirectionVector()->multiply($linearMaxDistance);
```

This solution is ludicrously inefficient.